dedupe/api.py

```diff
@@ -227,7 +233,10 @@ def pairs(self, data):
         # Blocking and pair generation are typically the first memory
         # bottlenecks, so we'll use sqlite3 to avoid doing them in memory
         with tempfile.TemporaryDirectory() as temp_dir:
-            con = sqlite3.connect(temp_dir + '/blocks.db')
+            if in_memory:
+                con = sqlite3.connect(':memory:')
```

Comment:
It doesn't seem like the worst thing, but it is a bit awkward that the temp_dir still gets created in this case where it is unused. It looks like there are some nice patterns for avoiding this, but some of them depend on python version >= 3.3 or 3.7. I'm not sure what dedupe is trying to support currently. See some of the answers on this stackoverflow: https://stackoverflow.com/questions/27803059/conditional-with-statement-in-python.

May not be worth worrying about the temp_dir at all.